### PR TITLE
Added 4x links to open/copy emote link menu.

### LINKED ISSUE #31: [Enhancement] Add 4x links on open/copy emote link menu for 7tv emotes.
**Describe your issue**

When right clicking on an emote to copy/open link on 7tv emotes:
 - only up to 3x shows, not up to 4x (the max)

**Screenshots**

Copy Link Section

**OS and Chatterino Version**

Chatterino Nightly 7.3.3 (commit 452da6bdd )

### PR DESCRIPTION
**Pull request by mmattbtw**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description
Added a button to open and/or copy the link to a 4x size of an emote. Closes #31 
image

### CODE REVIEW

**Comment by goldbattle** [comment]
Hi, to my understanding, there are a couple issues in general with larger scale emotes in chatterino. The current scale values don't make much sense to me, and should really be based off the vertical height of the expected image (e.g. 32, 64, 128 etc). 

To have constant scaling the ratios are a bit different and have a larger impact on the 4x resolution (for example right now 3x emotes are using the scale factor that 4x emotes should be using). Additionally, right now if a high resolution image is requested only the 3x image max will be returned from the `getImagePriv` function. Here is a commit which addresses these issues if you are interested: https://github.com/goldbattle/chatterino2/commit/403b947b6351f2c6bc6cf3f756e536cdfc00ab96

**Comment by mmattbtw** [comment]
Went ahead and cherry picked it into this branch, thank you :D

File: src/messages/ImageSet.cpp [hunk 1]
```diff
@@ -66,16 +81,21 @@ const std::shared_ptr<Image> &getImagePriv(const ImageSet &set, float scale)
...
         quality = 3;
     else if (scale > 1.001f)
         quality = 2;
 
-    if (!set.getImage3()->isEmpty() && quality == 3)
+    if (!set.getImage4()->isEmpty() && quality == 4)
+    {
+        return set.getImage4();
+    }
+    if (!set.getImage3()->isEmpty() && quality >= 3)
```
Comment (jammehcow) [inline src/messages/ImageSet.cpp:95]: Tiny nitpick on formatting
```suggestion
    }

    if (!set.getImage3()->isEmpty() && quality >= 3)
```

File: src/messages/ImageSet.cpp [hunk 2]
```diff
@@ -66,16 +81,21 @@ const std::shared_ptr<Image> &getImagePriv(const ImageSet &set, float scale)
...
     {
         return set.getImage3();
     }
-
```
Comment (jammehcow) [inline src/messages/ImageSet.cpp:78]: Same as suggestion above

**Review by jammehcow — APPROVED** [review]
Logic looks good, haven't tested it.

**Comment by jammehcow** [comment]
Tested ff3747d and works as expected.

**Comment by TroyKomodo** [comment]
Hi @jammehcow, please can you rebase this branch again. Sorry for the inconvenience thank you for your contributions.

File: src/providers/bttv/BttvEmotes.cpp [hunk 3]
```diff
@@ -58,7 +58,7 @@ namespace {
                 name,
                 ImageSet{Image::fromUrl(getEmoteLinkV3(id, "1x"), 1),
                          Image::fromUrl(getEmoteLinkV3(id, "2x"), 0.5),
-                         Image::fromUrl(getEmoteLinkV3(id, "3x"), 0.25)},
+                         Image::fromUrl(getEmoteLinkV3(id, "3x"), 0.286)},
```
Comment (TroyKomodo) [inline src/providers/bttv/BttvEmotes.cpp:61]: Why is this being changed?

Comment (mmattbtw) [reply src/providers/bttv/BttvEmotes.cpp:61]: I believe this was a scaling change from @goldbattle

If you want, I can revert the PR before that commit tomorrow, and then do scaling changes in a separate PR.

Comment (goldbattle) [reply src/providers/bttv/BttvEmotes.cpp:61]: This comment has details
https://github.com/SevenTV/chatterino7/pull/62#issuecomment-1008244983

I was planning on making a PR upstream to fix and discuss with them, but forgot about it.

Comment (TroyKomodo) [reply src/providers/bttv/BttvEmotes.cpp:61]: I am also curious where the numbers come from, why `0.286`?

Comment (TroyKomodo) [reply src/providers/bttv/BttvEmotes.cpp:61]: So also @mmattbtw this change is not from this pr? Is is an artifact from a merge?

Comment (mmattbtw) [reply src/providers/bttv/BttvEmotes.cpp:61]: I cherry-picked this commit into this PR. https://github.com/goldbattle/chatterino2/commit/403b947b6351f2c6bc6cf3f756e536cdfc00ab96

Comment (TroyKomodo) [reply src/providers/bttv/BttvEmotes.cpp:61]: Ah, i see it is. In future please dont use git merge, rather do a git rebase and then force push the branch.

**Comment by TroyKomodo** [comment]
@mmattbtw Thank you for your contributions!
In future please dont use `git merge` rather use `git rebase` and then force push your branch.

**Comment by TroyKomodo** [comment]
@mmattbtw Feel free to tag me on discord if you do not have the contributor role.

**Pull request merged**